src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -8,76 +8,107 @@ import * as telemetry from '../../shared/telemetry/telemetry'
 import {
     CloudWatchLogsData,
     CloudWatchLogsGroupInfo,
-    CloudWatchLogsParameters,
     LogStreamRegistry,
     filterLogEventsFromUriComponents,
+    CloudWatchLogsParameters,
 } from '../registry/logStreamRegistry'
 import { DataQuickPickItem } from '../../shared/ui/pickerPrompter'
 import { Wizard } from '../../shared/wizards/wizard'
 import { createInputBox } from '../../shared/ui/inputPrompter'
 import { createURIFromArgs } from '../cloudWatchLogsUtils'
 import { DefaultCloudWatchLogsClient } from '../../shared/clients/cloudWatchLogsClient'
 import { CloudWatchLogs } from 'aws-sdk'
+import { LogGroupNode } from '../explorer/logGroupNode'
 import { RegionSubmenu, RegionSubmenuResponse } from '../../shared/ui/common/regionSubmenu'
 import { highlightDocument } from '../document/logStreamDocumentProvider'
 import { CancellationError } from '../../shared/utilities/timeoutUtils'
 import { localize } from 'vscode-nls'
 import { getLogger } from '../../shared/logger'
 
-export async function searchLogGroup(registry: LogStreamRegistry): Promise<void> {
-    let result: telemetry.Result = 'Succeeded'
-    const response = await new SearchLogGroupWizard().run()
-    if (response) {
-        const logGroupInfo: CloudWatchLogsGroupInfo = {
-            groupName: response.submenuResponse.data,
-            regionName: response.submenuResponse.region,
-        }
+function handleWizardResponse(response: SearchLogGroupWizardResponse, registry: LogStreamRegistry): CloudWatchLogsData {
+    const logGroupInfo = {
+        groupName: response.submenuResponse.data,
+        regionName: response.submenuResponse.region,
+    }
+
+    const parameters: CloudWatchLogsParameters = {
+        limit: registry.configuration.get('limit', 10000),
+        filterPattern: response.filterPattern,
+    }
 
-        const parameters: CloudWatchLogsParameters = {
-            limit: registry.configuration.get('limit', 10000),
-            filterPattern: response.filterPattern,
+    const initialStreamData: CloudWatchLogsData = {
+        data: [],
+        parameters: parameters,
+        busy: false,
+        logGroupInfo: logGroupInfo,
+        retrieveLogsFunction: filterLogEventsFromUriComponents,
+    }
+    return initialStreamData
+}
+
+async function prepareDocument(uri: vscode.Uri, registry: LogStreamRegistry): Promise<void> {
+    const doc = await vscode.workspace.openTextDocument(uri) // calls back into the provider
+    vscode.languages.setTextDocumentLanguage(doc, 'log')
+
+    const textEditor = await vscode.window.showTextDocument(doc, { preview: false })
+    registry.setTextEditor(uri, textEditor)
+
+    // Initial highlighting of the document and then for any addLogEvent calls.
+    highlightDocument(registry, uri)
+    vscode.workspace.onDidChangeTextDocument((event: vscode.TextDocumentChangeEvent) => {
+        if (event.document.uri.toString() === doc.uri.toString()) {
```

Comment:
Is this a stable way to compare vscode.Uris? do we need something similar to `pathUtils.areEqual()` ?

related: https://github.com/microsoft/vscode/issues/93368